eth/stagedsync/stage_mining_create_block.go

```diff
@@ -115,7 +116,26 @@ func SpawnMiningCreateBlockStage(s *StageState, tx kv.RwTx, cfg MiningCreateBloc
 	}
 
 	if cfg.blockBuilderParameters != nil && cfg.blockBuilderParameters.ParentHash != parent.Hash() {
-		return fmt.Errorf("wrong head block: %x (current) vs %x (requested)", parent.Hash(), cfg.blockBuilderParameters.ParentHash)
+		if cfg.chainConfig.IsOptimism() {
+			log.Warn("wrong head block", "current", parent.Hash(), "requested", cfg.blockBuilderParameters.ParentHash, "executionAt", executionAt)
+			exectedParent, err := rawdb.ReadHeaderByHash(tx, cfg.blockBuilderParameters.ParentHash)
+			if err != nil {
+				return err
+			}
+			executionAt = exectedParent.Number.Uint64()
+			err = s.Update(tx, executionAt)
+			if err != nil {
+				return err
+			}
+			parent = exectedParent
+
```

Comment:
Suggestion: Remove `err = s.Update(tx, executionAt)` and make stage progress update uniform, only using `stage.SaveStageProgress`.`Update()` is a wrapper for `stages.SaveStageProgress(db, s.ID, newBlockNum)`.